lib/poseidon/broker_pool.rb

```diff
@@ -65,8 +65,15 @@ def close
     private
     def fetch_metadata_from_broker(broker, topics)
       host, port = broker.split(":")
-      Connection.open(host, port, @client_id, @socket_timeout_ms, @connect_timeout_ms) do |connection|
+      port = port.to_i
```

Comment:
Are you sure you need this?

I'm not seeing anything in connection.rb that turns the port param into an int from a string...